exec/exploratory_plots.R

```diff
@@ -179,15 +185,57 @@ if (is.null(opt$final_assay)){
   }
 }
 
-# Guess which assays aren't logged and log them. Don't like this, but need a
-# cleaner solution
+if (is.null(opt$log2_assays)) {
+
+  print("--log2_assays param not set, will guess which input assays need to be log2.")
+  unlogged <- unlist(lapply(assay_data, function(x) max(x) > 20))
+  for (unlogged_expression_type in names(unlogged)[unlogged]) {
+    if (unlogged_expression_type %in% names(assay_data)) {
+      assay_data[[unlogged_expression_type]] <- log2(assay_data[[unlogged_expression_type]])
+    }
+  }
+
+} else if (opt$log2_assays == "") {
+
+  print("--log2_assays param set to empty string, will not apply log2.")
 
-unlogged <- unlist(lapply(assay_data, function(x) max(x) > 20))
+} else {
+  if (is_valid_positive_integer_vector(opt$log2_assays)) {
 
-for (unlogged_expression_type in names(unlogged)[unlogged]) {
-  if (unlogged_expression_type %in% names(assay_data)) {
-    assay_data[[unlogged_expression_type]] <- log2(assay_data[[unlogged_expression_type]])
+    print("--log2_assays param set to list of int, will apply log2 to specified assays.")
+    unlogged <- unique(as.integer(simpleSplit(opt$log2_assays)))
+
+    # Check if log2_assays contains 0
+    if (min(unlogged) == 0) {
+
+      stop(paste("0 is not a valid assay position, use 1-based positions. Please check param --log2_assays."))
+    }
+    # Check if highest position in log2_assays is valid
+    if (max(unlogged) > length(assay_data)) {
+
+      stop(paste(max(unlogged), "is not a valid assay position, only got", length(assay_data), "assay(s). Please check param --log2_assays."))
+    }
```

Comment:
```suggestion
    unlogged <- unique(as.integer(simpleSplit(opt$log2_assays)))
    
    invalid_assays <- unlogged[! unlogged %in% 1:length(assay_data)]
    if (length(invalid_assays) > 0){
        stop(paste0("Invalid assays: ", paste(invalid_assays, collapse=', ')))
    }
```

Sorry, I still think this is the better way.

Reply:
I'm sorry, I confused something and did not implement all your suggested changes from earlier.